pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,720 @@
+    def _add_frequency_sweep(self, sweep_data):
+        """Add a frequency sweep.
+
+        Parameters
+        ----------
+        sweep_data: EdbFrequencySweep
+        """
+        self._sweep_list[sweep_data.name] = sweep_data
+        edb_setup_info = self.get_sim_setup_info
+
+        if self._setup_type in ["kSIwave", "kHFSS"]:
+            for _, v in self._sweep_list.items():
+                edb_setup_info.SweepDataList.Add(v._edb_object)
+
+        self._edb_object = self._set_edb_setup_info(edb_setup_info)
+        self._update_setup()
+
+    @pyaedt_function_handler
+    def delete_frequency_sweep(self, sweep_data):
+        """Delete a frequency sweep.
+
+        Parameters
+        ----------
+        sweep_data : EdbFrequencySweep
+        """
+        name = sweep_data.name
+        if name in self._sweep_list:
+            self._sweep_list.pop(name)
+
+        fsweep = []
+        for k, val in self.frequency_sweeps.items():
+            if not k == name:
+                fsweep.append(val)
+        self.get_sim_setup_info.SweepDataList.Clear()
+        for i in fsweep:
+            self.get_sim_setup_info.SweepDataList.Add(i._edb_object)
+        self._update_setup()
+        return True if name in self.frequency_sweeps else False
+
+    @pyaedt_function_handler()
+    def add_frequency_sweep(self, name=None, frequency_sweep=None):
+        """Add frequency sweep.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the frequency sweep.
+        frequency_sweep : list, optional
+            List of frequency points.
+
+        Returns
+        -------
+        :class:`pyaedt.edb_core.edb_data.simulation_setup_data.EdbFrequencySweep`
+
+        Examples
+        --------
+        >>> setup1 = edbapp.create_siwave_syz_setup("setup1")
+        >>> setup1.add_frequency_sweep(frequency_sweep=[
+        ...                           ["linear count", "0", "1kHz", 1],
+        ...                           ["log scale", "1kHz", "0.1GHz", 10],
+        ...                           ["linear scale", "0.1GHz", "10GHz", "0.1GHz"],
+        ...                           ])
+        """
+        if name in self.frequency_sweeps:
+            return False
+
+        if not frequency_sweep:
+            frequency_sweep = [["linear scale", "0.1GHz", "10GHz", "0.1GHz"]]
+
+        if not name:
+            name = generate_unique_name("sweep")
+        sweep = EdbFrequencySweep(self, frequency_sweep, name)
+        self._add_frequency_sweep(sweep)
+        self._update_setup()
+        return sweep
+
+
+class EdbFrequencySweep(object):
+    """Manages EDB methods for frequency sweep.
+
+    Parameters
+    ----------
+    sim_setup : :class:`pyaedt.edb_core.edb_data.siwave_simulation_setup_data.SiwaveSYZSimulationSetup`
+    name : str, optional
+        Name of the frequency sweep.
+    edb_sweep_data : :class:`Ansys.Ansoft.Edb.Utility.SIWDCIRSimulationSettings`, optional
+        Edb object.
```

Comment:
```suggestion
        EDB object. The default is ``None``.
```
You should describe the behavior that occurs when ``None`` is used: The default is ``None``, in which case ....